internal/events/event.go

```diff
@@ -314,14 +317,30 @@ func (e Events) queryEvents(ctx context.Context, selectedCategories []string, cl
 	}
 
 	if result == nil { //non supported option
-		return make([]*common.Event, 0), nil
+		return make([]*common.Event, 0), swag.Int64(0), nil
 	}
 
 	var events []*common.Event
-	return events, WithIDs(result).Find(&events).Error
+
+	result = WithIDs(result)
+	totalRecords := result.Find(&events).RowsAffected
```

Comment:
you don't really want the events here just the count
```suggestion
    var totalRecords int64 
    result.Count(totalRecords)
```

Reply:
It doesn't work for me, not sure why. This is the only way I could find that gets the accurate count @eranco74

Reply:
I thought it would be good for returning nil when an error occurs, WDYT?

Reply:
It doesn't work for me, not sure why. This is the only way I could find that gets the accurate count @eranco74